specification/communication/data-plane/Microsoft.CommunicationServicesAdministration/ga/2021-03-07/phonenumbers.json

```diff
@@ -0,0 +1,932 @@
+        "callbackUrl": {
+          "description": "The webhook URL on the phone number configuration.",
+          "type": "string"
+        },
+        "applicationId": {
+          "description": "The application id of the application to which to configure",
+          "type": "string"
+        }
+      }
+    },
+    "AcquiredPhoneNumbers": {
+      "description": "A wrapper around a list of acquired phone numbers",
+      "type": "object",
+      "properties": {
+        "phoneNumbers": {
+          "description": "Represents a list of phone numbers",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AcquiredPhoneNumber"
+          }
+        },
+        "nextLink": {
+          "description": "Represents the URL link to the next page of phone number results",
+          "type": "string"
+        }
+      }
+    },
+    "PhoneNumberLongRunningOperation": {
+      "description": "Represents the status of a long running operation",
+      "type": "object",
+      "properties": {
+        "status": {
+          "description": "Represents the status of a long running operation",
+          "enum": [
+            "NotStarted",
+            "Running",
+            "Succeeded",
+            "Failed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PhoneNumberLongRunningOperationStatusCode",
+            "modelAsString": true
+          }
+        },
+        "resourceLocation": {
+          "type": "string"
+        },
+        "createdDateTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "error": {
+          "$ref": "#/definitions/CommunicationError"
+        },
+        "id": {
+          "type": "string"
+        },
+        "longRunningOperationType": {
+          "description": "Represents a category of long running operation",
+          "enum": [
+            "Purchase",
+            "ReleasePhoneNumber",
+            "Search",
+            "UpdatePhoneNumberCapabilities"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "LongRunningOperationType",
+            "modelAsString": true
+          }
+        },
+        "lastActionDateTime": {
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "SearchRequest": {
+      "description": "Represents a phone number search request.",
+      "required": [
+        "assignmentType",
+        "capabilities",
+        "phoneNumberType"
+      ],
+      "type": "object",
+      "properties": {
+        "phoneNumberType": {
+          "description": "The phone number type of the search.",
+          "enum": [
+            "Geographic",
+            "TollFree",
+            "Unknown"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PhoneNumberType",
+            "modelAsString": true
+          }
+        },
+        "assignmentType": {
```

Comment:
Same here, let's extract this as its own model.

https://github.com/Azure/azure-rest-api-specs/blob/3e42c16fc1fbfaaa5b236c88371bfb53dd34175d/specification/communication/data-plane/Microsoft.CommunicationServicesAdministration/preview/2020-11-01-preview3/phonenumbers.json#L486-L497

Reply:
As discussed offline, will leave as-is since Swashbuckle can't generate common parameter easily.